/*
 * Copyright 2012-2017 Johns Hopkins University
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * Developed by: Roberts Group
 *               Johns Hopkins University
 *               http://biophysics.jhu.edu/roberts/
 *
 * Author(s): Elijah Roberts, Max Klein
 */
syntax = "proto2";
package lm.io;

import "lm/io/FirstPassageTimes.proto";
import "lm/io/OrderParameterFirstPassageTimes.proto";
import "lm/io/OrderParametersValues.proto";
import "lm/io/SpeciesCounts.proto";
import "robertslab/pbuf/NDArray.proto";

message CMEState {
    required uint64 total_steps                                                     = 3;
    required SpeciesCounts species_counts                                           = 1;
    optional robertslab.pbuf.NDArray degree_advancements                            = 2;    //1D; rows=species; type=uint64
    optional OrderParametersValues order_parameter_values                           = 31;
    repeated FirstPassageTimes first_passage_times                                  = 11;
    repeated OrderParameterFirstPassageTimes order_parameter_first_passage_times    = 21;

    // The histograms storing the accumulated time for the tilings.
    repeated robertslab.pbuf.NDArray tiling_histogram                               = 40;

    repeated uint64 barrier_crossings                                               = 50;
}
